lib/pbench/test/unit/server/test_schema.py

```diff
@@ -263,7 +281,7 @@ def test_invalid_keyword_type(self, test):
         assert str(test) in str(exc)
 
     @pytest.mark.parametrize(
-        "test", ("I'm not sure", "yes!", "ebyam"),
+        "test", ("I'm not sure", "yes!", "ebyam", "yes.foo", "yes."),
```

Comment:
Now I'm thinking about whether I should add additional overhead above simply lower/split to evaluate incoming keywords. Again, only the open namespace is a potential issue because everything else has to match the actually structure of the JSON documents (and the primary key). But `user..foo.` probably should be rejected.

Reply:
> `user..foo.` probably should be rejected.

I think it will be rejected (at least, on the query side), because of the two null key values, without any changes to your code.  The one change to make that work is on the create side, to insist that key values be non-null, and then we've got it covered.

The point of my question was that I think a query on `yes.` will (or should) always be invalid, regardless of how the key was defined.